sentry_sdk/integrations/litestar.py

```diff
@@ -0,0 +1,282 @@
+from sentry_sdk.integrations.logging import ignore_logger
+from sentry_sdk.scope import Scope as SentryScope, should_send_default_pii
+from sentry_sdk.tracing import SOURCE_FOR_STYLE, TRANSACTION_SOURCE_ROUTE
+from sentry_sdk.utils import (
+    ensure_integration_enabled,
+    event_from_exception,
+    transaction_from_function,
+)
+
+try:
+    from litestar import Request, Litestar  # type: ignore
+    from litestar.handlers.base import BaseRouteHandler  # type: ignore
+    from litestar.middleware import DefineMiddleware  # type: ignore
+    from litestar.routes.http import HTTPRoute  # type: ignore
+    from litestar.data_extractors import ConnectionDataExtractor  # type: ignore
+
+    if TYPE_CHECKING:
+        from typing import Any, Dict, List, Optional, Union
+
+        from litestar.types.asgi_types import ASGIApp  # type: ignore
+        from litestar.types import (  # type: ignore
+            HTTPReceiveMessage,
+            HTTPScope,
+            Message,
+            Middleware,
+            Receive,
+            Scope as LitestarScope,
+            Send,
+            WebSocketReceiveMessage,
+        )
+        from litestar.middleware import MiddlewareProtocol
+        from sentry_sdk._types import Event, Hint
+except ImportError:
+    raise DidNotEnable("Litestar is not installed")
+
+
+_DEFAULT_TRANSACTION_NAME = "generic Litestar request"
+
+
+class LitestarIntegration(Integration):
+    identifier = "litestar"
+    origin = f"auto.http.{identifier}"
+
+    @staticmethod
+    def setup_once() -> None:
+        patch_app_init()
+        patch_middlewares()
+        patch_http_route_handle()
+
+        # logs an error for every 500
+        ignore_logger("litestar")
+
+
+class SentryLitestarASGIMiddleware(SentryAsgiMiddleware):
+    def __init__(self, app: "ASGIApp", span_origin: str = LitestarIntegration.origin):
+        super().__init__(
+            app=app,
+            unsafe_context_data=False,
+            transaction_style="endpoint",
+            mechanism_type="asgi",
+            span_origin=span_origin,
+        )
+
+
+def patch_app_init() -> None:
+    """
+    Replaces the Litestar class's `__init__` function in order to inject `after_exception` handlers and set the
+    `SentryLitestarASGIMiddleware` as the outmost middleware in the stack.
+    See:
+    - https://docs.litestar.dev/2/usage/applications.html#after-exception
+    - https://docs.litestar.dev/2/usage/middleware/using-middleware.html
+    """
+    old__init__ = Litestar.__init__
+
+    def injection_wrapper(self: "Litestar", *args: "Any", **kwargs: "Any") -> None:
+        after_exception = kwargs.pop("after_exception", [])
+        kwargs.update(
+            after_exception=[
+                exception_handler,
+                *(
+                    after_exception
+                    if isinstance(after_exception, list)
+                    else [after_exception]
+                ),
+            ]
+        )
```

Comment:
We can simplify this like so:

```suggestion
        kwargs["after_exception"] = [
            exception_handler,
            *kwargs.get("after_exception", []),
        ]
```

The simplification also avoids potentially modifying the `after_exception` list, as in your previous suggestion. `Litestar`'s decision to type `after_exception` as the immutable `Sequence` suggests to users that the value will not be modified, so we should respect that in our integration.